gef-ui/server/controllers/about-facility/index.js

```diff
@@ -38,321 +47,94 @@ const aboutFacility = async (req, res) => {
-    if (coverEndDate < coverStartDate && coverEndDateValid) {
-      aboutFacilityErrors.push({
-        errRef: 'coverEndDate',
-        errMsg: 'Cover end date cannot be before cover start date',
-      });
-    }
-
-    // if coverEndDate is the same as the coverStartDate
-    if (isEqual(coverStartDate, coverEndDate) && coverEndDateValid) {
-      aboutFacilityErrors.push({
-        errRef: 'coverEndDate',
-        errMsg: 'The cover end date must be after the cover start date',
-      });
-    }
-  }
-
-  // validation if should start on submission - no coverStartDate to check against
-  if (body.shouldCoverStartOnSubmission && coverEndDateIsFullyComplete) {
-    // temporarily set coverStartDate to now
-    const coverStartNow = set(new Date(), {
-      hours: 0,
-      minutes: 0,
-      seconds: 0,
-      milliseconds: 0,
-    });
-
-    if (coverEndDate < coverStartNow && coverEndDateValid) {
-      aboutFacilityErrors.push({
-        errRef: 'coverEndDate',
-        errMsg: 'Cover end date cannot be before cover start date',
-      });
-    }
-
-    if (isEqual(coverStartNow, coverEndDate) && coverEndDateValid) {
-      aboutFacilityErrors.push({
-        errRef: 'coverEndDate',
-        errMsg: 'The cover end date must be after the cover start date',
-      });
-    }
-  }
-
-  // Regex tests to see if value is a number only
-  const digitsRegex = /^[0-9]*$/;
-  if (body.monthsOfCover) {
-    if (!digitsRegex.test(body.monthsOfCover)) {
-      aboutFacilityErrors.push({
-        errRef: 'monthsOfCover',
-        errMsg: 'You can only enter numbers',
-      });
-    }
-    if (body.monthsOfCover > 999) {
-      aboutFacilityErrors.push({
-        errRef: 'monthsOfCover',
-        errMsg: 'You can only enter a maximum of 999 months cover',
-      });
-    }
-  }
+  const coverStartDate = coverStartDateIsFullyComplete
+    ? set(new Date(), {
+        year: coverStartDateYear,
+        month: coverStartDateMonth - 1,
+        date: coverStartDateDay,
+        hours: 0,
+        minutes: 0,
+        seconds: 0,
+        milliseconds: 0,
+      })
+    : null;
+  const coverEndDate = coverEndDateIsFullyComplete
+    ? set(new Date(), {
+        year: coverEndDateYear,
+        month: coverEndDateMonth - 1,
+        date: coverEndDateDay,
+        hours: 0,
+        minutes: 0,
+        seconds: 0,
+        milliseconds: 0,
+      })
+    : null;
+
+  const aboutFacilityErrors = validateAboutFacility({
+    coverStartDateDay,
+    coverStartDateMonth,
+    coverStartDateYear,
+    coverStartDateIsFullyComplete,
+    coverEndDateDay,
+    coverEndDateMonth,
+    coverEndDateYear,
+    coverEndDateIsFullyComplete,
+    facilityTypeString,
+    saveAndReturn,
+    coverStartDate,
+    coverEndDate,
+    hasBeenIssued,
+    monthsOfCover,
+    facilityName,
+    shouldCoverStartOnSubmission,
+    facilityEndDateExists,
+    isFacilityEndDateEnabled: isFacilityEndDateEnabledOnDeal(deal.version ?? 0),
```

Comment:
I've made a function to commonise this in libs/common as I expect this will be done on lots of other pages too & these pages shouldn't know about how the versioning is implemented.